plugin/src/main/java/com/denizenscript/denizen/objects/NPCTag.java

```diff
@@ -1307,6 +1369,101 @@ else if (attribute.startsWith("list", 2)) {
                 hologram.addTemporaryLine(text.asString(), duration.getTicksAsInt());
             }
         });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name waypoint_provider
+        // @input ElementTag
+        // @description
+        // Sets the NPC's Waypoints Provider (default options: linear, wander, guided.)
+        // @tags
+        // <NPCTag.waypoint_provider>
+        // -->
+        tagProcessor.registerMechanism("waypoint_provider", false, ElementTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            wp.setWaypointProvider(input.asString());
+        });
+
+        // <--[mechanism]
+        // @object DurationTag
+        // @name wander_delay
+        // @input DurationTag
+        // @description
+        // Sets the delay (in seconds) for an NPC's wander Waypoints Provider. Provider must be set to Wander before setting wander_delay. Set to -1 to use Mincraft's default value.
+        // @tags
+        // <NPCTag.wander_delay>
+        // -->
+        tagProcessor.registerMechanism("wander_delay", false, DurationTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            if (wp.getCurrentProvider() instanceof WanderWaypointProvider wanderWaypointProvider) {
+                if (input.getSecondsAsInt() >= -1) {
+                    wanderWaypointProvider.setDelay(input.getSecondsAsInt());
+                }
+                else {
+                    mechanism.echoError("Invalid value for wander delay");
+                }
+            }
+        });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name wander_worldguardregion
+        // @input ElementTag
+        // @description
+        // Sets the worldguardRegion for an NPC's wander Waypoints Provider.  
```

Comment:
shouldn't be here - should be left for Depenizen to fill with a WorldguardRegionTag

Reply:
This was the part I was most interested in!